tests/transforms/test_batch_transform.py

```diff
@@ -605,3 +605,99 @@ def cost(x, weights):
         grad = jax.grad(cost, argnums=[0, 1])(x, weights)
         expected = qml.grad(self.expval)(np.array(x), np.array(weights))
         assert all(np.allclose(g, e) for g, e in zip(grad, expected))
+
+    def test_batch_transforms_qnode(self, diff_method, mocker):
+        """Test that batch transforms can be applied to a QNode
+        without affecting device batch transforms"""
+        if diff_method == "backprop":
+            pytest.skip("Test only supports finite shots")

```

Comment:
To confirm, this restriction is an implementation detail of the test itself rather than a limitation on qnode batch plus dev batch transforms?

Reply:
Yep, that's correct. Currently, the only `device.batch_transform` that is applied is the one that expands Hamiltonians if `shots!=None` --- but this scenario can't be done with backprop.

I considered mocking `device.batch_transform` to do a test-specific transform, do you think this should be added?